app/controllers/katello/concerns/smart_proxies_controller_extensions.rb

```diff
@@ -0,0 +1,45 @@
+module Katello
+  module Concerns
+    module SmartProxiesControllerExtensions
+      extend ActiveSupport::Concern
+
+      included do
+        append_view_path('app/views/foreman')
+        before_filter :find_resource_and_status, :only => [:pulp_storage, :pulp_status]
+        alias_method_chain :action_permission, :katello
+
+        def pulp_storage
+          @storage = @proxy_status[:pulp].storage
+          respond_to do |format|
+            format.html { render :layout => false }
+            format.json { render :json => {:success => true, :message => @storage} }
+          end
+        end
+
+        def pulp_status
+          @pulp_status = @proxy_status[:pulp].status
```

Comment:
When storage cannot be detected, spinner spins indefinitely. Maybe we could handle the exception here and render appropriate json.